### PR TITLE
(frozen) scroll dev 0818

### PR DESCRIPTION
**Pull request by lispc**
Direct push forbidden.  You have to make a PR against this branch.

### CODE REVIEW

**Comment by lispc** [comment]
@DreamWuGit do you really need `bytecode_circuit_pad_to`?

File: zkevm-circuits/src/evm_circuit/execution.rs [hunk 1]
```diff
@@ -755,6 +755,51 @@ impl<F: Field> ExecutionConfig<F> {
...
+                                .assign_advice(|| "assign advice rows", column, i, || Ok(F::zero()))
+                                .unwrap();
+                            region
+                                .assign_fixed(
+                                    || "assgin fixed q_usable rows ",
+                                    self.q_usable,
+                                    i,
+                                    || Ok(F::one()),
+                                )
+                                .unwrap();
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/execution.rs:789]: @DreamWuGit  you assign q_usable multi times inside each outer for loop. Better refactor the loops like this

```
for i in 0..end_row 
  assign_fixed q_usable
  for column in xxx:
    assign_advice column
     
```

Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/execution.rs:789]: yeah, will refactor it !!

Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/execution.rs]: updated it [59e83bd](https://github.com/scroll-tech/zkevm-circuits/pull/186/commits/59e83bd79bb504ea3437e88b513b31dbb4a83828) , I am thinking if all multi assigning can be regarding as errors , if yes, halo2 should check it and reports this error at the low level (a possible issue to halo2)

Comment (DreamWuGit) [reply zkevm-circuits/src/evm_circuit/execution.rs]: @lispc  open an issue regarding this in halo2 for discussion https://github.com/zcash/halo2/issues/645

**Comment by lispc** [comment]
oh https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/654 makes a lot of refactor for bytecode circuit ..

**Pull request closed without merging**